tests/books_api_tests.py

```diff
@@ -0,0 +1,102 @@
+# api endpoint tests
+# path ../tests/book_api_tests.py
+
+# import os and python unittest modules
+
+import os
+import unittest
+
+# import json
+import json
+
+from app import create_app
+
+
+class HellobooksCRUDTestCase(unittest.TestCase):
+    '''This class represents the Hello_Books_Api Testcase'''
+
+    def setup(self):
+        '''Define test variables and initialize app.'''
+        self.app = create_app('testing')
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+
+        # book(dict) variable
+        self.book_details = {
+            'id': 4,
+            'title':'Eloquent Javascript',
+            'edition': 'Second',
+            'description': '',
+            'author': 'Marijn Haverbeke',
+            'category':'Javascript',
+            'copies': 20
+
+        }
+
+
+    def tearDown(self):
+        self.app_context.pop()
+
+
+    def test_book_info_can_be_modified(self):
+        '''test whether book info can be updated'''
+        result_post=self.client.post('/api/books/', data = self.book_details)
+        self.assertEqual(result_post.status_code, 201)
+        new_book_detail = self.book_details['description'] = "A modern introduction to programming"
+        result_put = self.client.put('/api/books/4', data = new_book_detail)
```

Comment:
I can't see where you are asserting the values of the put request.

Reply:
ooh yeah, let me fix that @justmesam